cmd/staticcheck/docs/checks/SA9004

```diff
@@ -0,0 +1 @@
+Using == or != operators to compare time.Time is problematic; prefer Time.Equal
```

Comment:
Needs a paragraph explaining *why* time.Time.Equal should be used instead, but I'll write that myself.